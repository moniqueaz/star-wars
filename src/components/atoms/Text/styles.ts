import styled, { css, DefaultTheme } from 'styled-components';

type WrapperProps = { color: 'light' | 'dark', size: 'small' | 'medium' | 'large' }

const wrapperModifiers = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  light: (theme: DefaultTheme) => css`color: ${theme.colors.white}`,
  dark: (theme: DefaultTheme) => css`color: ${theme.colors.primary}`,
};

export const Wrapper = styled.p<WrapperProps>`
  ${( { theme, color, size } ) => css`
      width: 100%;
      ${wrapperModifiers[color](theme)};
      ${wrapperModifiers[size](theme)};
  `}
`;
