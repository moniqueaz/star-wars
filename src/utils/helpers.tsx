export const fetcher = (url: string) => fetch(url).then(res => res.json());

export const getRandomInteger = (min: number, max: number) => {
  return Math.floor(Math.random() * (max - min)) + min;
};
