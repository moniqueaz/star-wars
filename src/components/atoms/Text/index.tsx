import * as S from './styles';

type TextProp = {
  children: React.ReactNode
  light?: boolean
  size?: 'small' | 'medium' | 'large'
}

const Text = ( { children, light = false, size = 'medium' }: TextProp) => {
  if (!children) {
    return null;
  }

  return (
    <S.Wrapper light={light} size={size}>
      {children}
    </S.Wrapper>
  );
};

export default Text;
