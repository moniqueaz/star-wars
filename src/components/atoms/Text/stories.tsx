import { Story, Meta } from '@storybook/react/types-6-0';
import Text from '.';

export default {
  title: 'Atoms/Text',
  component: Text,
} as Meta;

export const Default: Story = args => {
  return <Text {...args} >Texto do paragrafo</Text>;
};

export const Dark: Story = () => {
  return <Text color="dark">Texto do paragrafo</Text>;
};

Dark.parameters = {
  backgrounds: {
    default: 'light',
  },
};
