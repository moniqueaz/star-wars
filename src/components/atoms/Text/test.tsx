import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import Text from '.';

describe('<Text />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Text data-testid="Text">Texto do paragrafo</Text>);
    expect(screen.getByText('Texto do paragrafo')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  } );
} );
