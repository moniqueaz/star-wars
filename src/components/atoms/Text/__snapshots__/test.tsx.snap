// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render 1`] = `
.c0 {
  width: 100%;
  color: #1A4173;
  font-size: 1.6rem;
}

<p
  class="c0"
>
  Texto do paragrafo
</p>
`;

exports[`<Text /> should render text large 1`] = `
.c0 {
  width: 100%;
  color: #1A4173;
  font-size: 1.8rem;
}

<p
  class="c0"
>
  Texto do paragrafo
</p>
`;

exports[`<Text /> should render text light 1`] = `
.c0 {
  width: 100%;
  color: #f1f1f1;
  font-size: 1.6rem;
}

<p
  class="c0"
>
  Texto do paragrafo
</p>
`;

exports[`<Text /> should render text medium 1`] = `
.c0 {
  width: 100%;
  color: #1A4173;
  font-size: 1.6rem;
}

<p
  class="c0"
>
  Texto do paragrafo
</p>
`;

exports[`<Text /> should render text small 1`] = `
.c0 {
  width: 100%;
  color: #1A4173;
  font-size: 1.4rem;
}

<p
  class="c0"
>
  Texto do paragrafo
</p>
`;
